### PR TITLE
feat: create TopReaderBadgeModal

### PR DESCRIPTION
**Pull request by omBratteng**
## Changes

Depends on https://github.com/dailydotdev/daily-api/pull/2376

TODO:

- [ ] Download image

## Events

Did you introduce any new tracking events?

https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit?gid=1801144038#gid=1801144038

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

AS-662

### Preview domain
https://as-662-top-reader-modal.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    return null;
+  }
+
+  return (
+    <Modal
+      {...props}
+      size={ModalSize.Small}
+      isDrawerOnMobile
+      onAfterClose={() => onAfterClose('keyword.value')}
+      onAfterOpen={() => onAfterOpen('keyword.value')}
```
Comment (rebelchris) [inline packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx:88]: Are these correct?

File: packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+          variant={ButtonVariant.Primary}
+          icon={<DownloadIcon secondary />}
+          disabled={!topReaderBadge.image}
+          loading={downloading}
+          onClick={() => onClickDownload()}
+        >
+          Download badge
+        </Button>
+
+        {!isMobile && (
```
Comment (rebelchris) [inline packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx:111]: Why do we need this specific condition, the modals have their generic use-case for mobile/desktop buttons already right?

**Review by rebelchris — CHANGES_REQUESTED** [review]
Minor questions

File: packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx (hunk 1 above)
Comment (omBratteng) [reply packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx:88]: no 😬

File: packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx (hunk 2 above)
Comment (omBratteng) [reply packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx:111]: [In the design](https://www.figma.com/design/nmfWPS7x3kzLUvYMkBx2kW/daily.dev---Dev-Mode?node-id=14437-23460&node-type=instance&m=dev), the close button is at the bottom. Usually they're a X button at the top right. And when it's on mobile, it already comes with close button on the drawer, so I need to hide it.

File: packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx (hunk 1 above)
Comment (omBratteng) [reply packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx]: Fixed

File: packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx (hunk 2 above)
Comment (omBratteng) [reply packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx:111]: I can set `displayCloseButton: false` on `drawerProps` and then always render this button instead.

Comment (omBratteng) [reply packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx:111]:
```diff
diff --git a/packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx b/packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx
index e535994f..578d9a7d 100644
--- a/packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx
+++ b/packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx
@@ -86,6 +86,9 @@ const TopReaderBadgeModal = (
       isDrawerOnMobile
       onAfterClose={() => onAfterClose(topReaderBadge.keyword.value)}
       onAfterOpen={() => onAfterOpen(topReaderBadge.keyword.value)}
+      drawerProps={{
+        displayCloseButton: false,
+      }}
     >
       <Modal.Body className="flex flex-col items-center justify-center gap-4 text-center">
         <h1 className="font-bold typo-title1">
@@ -108,15 +111,13 @@ const TopReaderBadgeModal = (
           Download badge
         </Button>
 
-        {!isMobile && (
-          <Button
-            className="w-full max-w-80"
-            variant={ButtonVariant.Float}
-            onClick={onRequestClose}
-          >
-            Close
-          </Button>
-        )}
+        <Button
+          className={classNames('w-full', !isMobile && 'max-w-80')}
+          variant={ButtonVariant.Float}
+          onClick={onRequestClose}
+        >
+          Close
+        </Button>
       </Modal.Body>
     </Modal>
   );
```

Comment (rebelchris) [reply packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx:111]: Yeah guess it's easier in case they change copy or something silly 🤪

**Review by rebelchris — APPROVED** [review]

File: packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx (hunk 2 above)
Comment (omBratteng) [reply packages/shared/src/components/modals/badges/TopReaderBadgeModal.tsx]: Changed to hide drawer button and use inline button instead.

**Pull request merged**